log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java

```diff
@@ -158,18 +160,18 @@ public void testRealProducerConfigWithSyncSendAndNotIgnoringExceptionsShouldThro
         assertThrows(RuntimeException.class, () -> logger.error(getMessage(0)));
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
Why do we need this?

Reply:
Because Mockito.mock() returns `MockProducer` but we need `MockProducer<byte[], byte[]>`, this is similar to https://github.com/apache/kafka/blob/5936a249bda171851cfa669da4276d23230eab6e/raft/src/test/java/org/apache/kafka/raft/internals/MemoryBatchReaderTest.java#L46

Reply:
Can we have the suppression at the variable level like it is in the example you mentioned?

Reply:
We have 2 variables here that need to be suppressed, `futureMock` and `producer`, so it seems better to suppress them at the method level, WDYT?

Reply:
No, it's still best to do it at the most specific place in my opinion.

Reply:
Thank you, Done!